js/bank.js

```diff
@@ -4,118 +4,113 @@ var rl = readline.createInterface({
   output: process.stdout
 });
 
-function start(balance) {
+
+function bank(balance) {
   this.balance = balance;
   var name = rl.question("Hello! Welcome to Citibank! What is you name? ", function(name) {
-    var decision = rl.question(name + ", would you like to: \n open an account, \n see your balance, \n make a withdrawal, \n make a deposit or \n close an account? ", function(decision){
-      if (decision == "balance") {
-        console.log("Your balance is $" + this.balance);
-        console.log("Goodbye, thanks for banking with Citibank!");
-        rl.close();
-      } else if (decision == "withdraw") {
-        this.balance = withdraw(this.balance);
-      } else if (decision == "deposit") {
-        this.balance = deposit(this.balance);
-      } else if (decision == "open an account") {
-        this.balance = open(this.balance);
-      } else if (decision == "close an account") {
-        this.balance = close(this.balance);
-      }
-      else {
-        var restart = rl.question("You did not make an appropriate decision, start from the beginning? Enter in yes, or no . " , function(restart) {
-          if (restart == "yes") {
-            start(this.balance);
-          }
-          else {
-            console.log("Goodbye, thanks for banking with Citibank!");
-             rl.close();
-            }
-        });
-      }
-    });
+    transaction(name, balance);
+  });
+}
+
+function transaction(name, balance) {
```

Comment:
I like that you added a function for transactions. 

Right now this function is still doing more than one thing.  You're passing `name` to a transaction.  It would be nice if you passed the selection made by the user too.  That way, the `transaction` isn't responsible for communicating with the user.

Reply:
I need the name because I acknowledge the user in line 16. I had decision earlier. I can update   that shortly

Reply:
It sounds like whatever is happening on line 16 should happen outside of the `transaction` function, and that data should be passed into the `transaction` function. That way, instead of having knowledge of how the `name` is created, that logic can be encapsulated into another function, and the result of that function can be passed to a `transaction`.

Reply:
I need line 16 because of the input(readline) (I attempted to create two separate functions and that did not work), however that I can remove the name as a variable.